zanata-war/src/main/java/org/zanata/action/LanguageJoinAction.java

```diff
@@ -42,119 +42,86 @@
+        "request_join_language";
 
     @In
-    private SendEmailAction sendEmail;
-    @In
     private Messages msgs;
 
     @In
     private LocaleMemberDAO localeMemberDAO;
 
-    @In(required = false, value = JpaIdentityStore.AUTHENTICATED_USER)
-    private HAccount authenticatedAccount;
-
-    @Getter
-    @Setter
-    private Boolean requestAsTranslator;
-
     @Getter
     @Setter
-    private Boolean requestAsReviewer;
+    private boolean requestAsTranslator;
 
     @Getter
     @Setter
-    private Boolean requestAsCoordinator;
+    private boolean requestAsReviewer;
 
     @Getter
     @Setter
-    private String emailType;
+    private boolean requestAsCoordinator;
 
     @Setter
     @Getter
     private String language;
 
-    private String title;
+    @In(value = JpaIdentityStore.AUTHENTICATED_USER, required = false)
+    private HAccount authenticatedAccount;
 
-    private String subject;
+    public String getEmailType() {
+        return EMAIL_TYPE_REQUEST_JOIN;
+    }
 
-    public boolean hasRoleRequest() {
+    public boolean hasSelectedRole() {
         return requestAsTranslator || requestAsReviewer || requestAsCoordinator;
     }
 
-    public String getSubject() {
-        if (emailType.equals(EMAIL_TYPE_REQUEST_JOIN)) {
-            subject =
-                    msgs.format("jsf.email.joinrequest.Subject",
-                            sendEmail.getFromLoginName(),
-                            sendEmail.getLocale().getLocaleId().getId());
-        } else {
-            subject =
-                    msgs.format("jsf.email.rolerequest.Subject",
-                            sendEmail.getFromLoginName(),
-                            sendEmail.getLocale().getLocaleId().getId());
+    public void bindRole(String role, boolean checked) {
+        if(role.equals("translator")) {
+            requestAsTranslator = checked;
+        } else if(role.equals("reviewer")) {
+            requestAsReviewer = checked;
+        } else if(role.equals("coordinator")) {
+            requestAsCoordinator = checked;
         }
-        return subject;
     }
 
-    public String getTitle() {
-        if (emailType.equals(EMAIL_TYPE_REQUEST_JOIN)) {
-            title =
-                    msgs
-                            .format("jsf.RequestToJoinLanguageTeamTitle",
-                                    sendEmail.getLocale().getLocaleId().getId());
-        } else {
-            title =
-                    msgs
-                            .format("jsf.RequestRoleLanguageTeamTitle",
-                                    sendEmail.getLocale().getLocaleId().getId());
+    public String getSubject() {
+        return msgs.format("jsf.language.email.joinrequest.Subject",
+            getLoginName(), getLocaleId().getId());
+    }
+
+    private String getLoginName() {
+        if(authenticatedAccount != null) {
+            return authenticatedAccount.getUsername();
         }
-        return title;
+        return "";
```

Comment:
This looks like someone could send a request to join a language team, even though we don't know who they are. I think if we can't get a username then it is pointless to send this email, and it should abort with an error.